### PR TITLE
{lib}[GCCcore 10.3.0,11.2.0,11.3.0,12.2.0,12.3.0,13.2.0] PSM2 v12.0.1

### PR DESCRIPTION
**Pull request by jfgrimm**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**FAILED**
Build succeeded for 1 out of 6 (5 easyconfigs in total)
node109.viking2.yor.alces.network - Linux Rocky Linux 8.9, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/7f2744bed6cac9a95f3f4364018002eb for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 1 out of 5 (5 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/d803f98df34c3397d4ad85adb73bb9b5 for a full test report.

**Comment by jfgrimm** [comment]
hmm, seems it doesn't like newer compilers

we've had it at SYSTEM-level at our site, but that won't make sense here because of the numactl runtime dep (we relied on RPATH and used a system-level numactl build dep instead)

**Comment by branfosj** [comment]
> hmm, seems it doesn't like newer compilers
> 
> we've had it at SYSTEM-level at our site, but that won't make sense here because of the numactl runtime dep (we relied on RPATH and used a system-level numactl build dep instead). I'll need to have a look at what's going wrong

Adding `binutils` as a build dep fixes that issue.

Also, please update patch with description.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/25bb214f143ebeecc242c074a8870c9e for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/p/PSM2/PSM2-12.0.1-GCCcore-10.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    ('CUDA', '11.3.1', '', SYSTEM),
+]
+
+dependencies = [
+    ('numactl', '2.0.14'),
+]
+
+skipsteps = ['configure']
+
+prebuildopts  = "sed -i 's|fprintf(stderr,|_HFI_DBG(|' psm_context.c && sed -i 's|/usr|/|' Makefile && "
```
Comment (branfosj) [inline easybuild/easyconfigs/p/PSM2/PSM2-12.0.1-GCCcore-10.3.0.eb:35]: Please remove one of the spaces in the `prebuildopts` line in each file.

```
prebuildopts  = "sed
            ^^
```

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**FAILED**
Build succeeded for 6 out of 8 (6 easyconfigs in total)
node048.viking2.yor.alces.network - Linux Rocky Linux 8.9, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/64d65780758a23dbea11b4645e2d1249 for a full test report.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 7 out of 7 (6 easyconfigs in total)
node048.viking2.yor.alces.network - Linux Rocky Linux 8.9, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/00a0356041292a0f81f228dec4819f1f for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/19989e1dfd3728608124871c352d30d2 for a full test report.

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Going in, thanks @jfgrimm!

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
jrc0870.jureca - Linux Rocky Linux 8.9, AArch64, ARM UNKNOWN (neoverse_n1), 2 x NVIDIA NVIDIA A100-PCIE-40GB, 535.154.05, Python 3.6.8
See https://gist.github.com/SebastianAchilles/0f8f6502c9600b8b6f6e3c6851e63236 for a full test report.

**Pull request merged**